generative/worley.glsl

```diff
@@ -17,42 +18,80 @@ license:
 #ifndef FNC_WORLEY
 #define FNC_WORLEY
 
+#ifndef WORLEY_JITTER
+#define WORLEY_JITTER 1.0
+#endif
+
 #ifndef WORLEY_DIST_FNC
 #define WORLEY_DIST_FNC distEuclidean
 #endif
 
-float worley(vec2 p){
+vec2 worley(vec2 p){
     vec2 n = floor( p );
     vec2 f = fract( p );
 
-    float dis = 1.0;
-    for( int j= -1; j <= 1; j++ )
+    float distF1 = 1.0;
+    float distF2 = 1.0;
+    vec2 off1, pos1;
+    vec2 off2, pos2;
+    for( int j= -1; j <= 1; j++ ) {
         for( int i=-1; i <= 1; i++ ) {	
-                vec2  g = vec2(i,j);
-                vec2  o = random2( n + g );
-                vec2  delta = g + o - f;
-                float d = WORLEY_DIST_FNC(g+o, f);
-                dis = min(dis,d);
+            vec2  g = vec2(i,j);
+            vec2  o = random2( n + g ) * WORLEY_JITTER;
+            vec2  p = g + o;
+            float d = WORLEY_DIST_FNC(p, f);
+            if (d < distF1) {
+                distF2 = distF1;
+                distF1 = d;
+                off2 = off1;
+                off1 = g;
+                pos2 = pos1;
+                pos1 = p;
+            }
+            else if (d < distF2) {
+                distF2 = d;
+                off2 = g;
+                pos2 = p;
+            }
+        }
     }
 
-    return 1.0-dis;
+    return vec2(distF1, distF2);
 }
 
```

Comment:
Worry a bit for back compatibility. Do you mind renaming this function from `vec2 worley(vec2 p)` to `vec2 worley2(vec2 p)`. Then creating a new `vec2 worley(vec2 p)` that just extracts the F1 of it?

```glsl
float worley(vec2 p) {
   return worley2(p).x;
}
```

Reply:
Done! 😊